src/symbol/shaping.ts

```diff
@@ -630,10 +715,18 @@ function shapeLines(shaping: Shaping,
         const positionedLine = {positionedGlyphs: [], lineOffset: 0};
         shaping.positionedLines[lineIndex] = positionedLine;
         const positionedGlyphs = positionedLine.positionedGlyphs;
-        let lineOffset = 0.0;
+
+        const {maxGlyphHeight, maxImageHeight} = calculateLineBlockMetrics(
+            glyphMap,
+            glyphPositions,
+            imagePositions,
+            line,
+            layoutTextSizeFactor
+        );
+        const maxLineHeight = Math.max(maxGlyphHeight, maxImageHeight);
 
         if (!line.length()) {
-            y += lineHeight; // Still need a line feed after empty line
+            y += maxLineHeight; // Still need a line feed after empty line
```

Comment:
Shouldn't be `lineHeight` here as before? For empty line we move by default line height, correct? When the line is empty, `maxLineHeight` is 0?